mesheryctl/internal/cli/root/perf/perf.go

```diff
@@ -129,79 +127,75 @@ var PerfCmd = &cobra.Command{
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
 		// Importing SMPS Configuration from the file
+		var req *http.Request
 		if filePath != "" {
-			var t models.PerformanceSpec
-			err := yaml.Unmarshal([]byte(filePath), &t)
+			smpsConfig, err := ioutil.ReadFile(filePath)
+
 			if err != nil {
-				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to unmarshal yaml file %s", filePath)))
-			}
-			if testDuration == "" {
-				testDuration = fmt.Sprintf("%fs", t.EndTime.Sub(t.StartTime).Seconds())
+				return err
 			}
-			if testURL == "" {
-				testURL = t.EndpointURL
-			}
-			if concurrentRequests == "" {
-				concurrentRequests = fmt.Sprintf("%d", t.Client.Connections)
+
+			req, err = http.NewRequest("POST", mctlCfg.GetPerf().LoadTestSmpsURI, bytes.NewBuffer(smpsConfig))
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to create new request to %s", mctlCfg.GetBaseMesheryURL())))
 			}
-			if qps == "" {
-				qps = fmt.Sprintf("%f", t.Client.Rps)
+		} else {
+
+			if testName == "" {
+				log.Debug("Test Name not provided")
+				testName = StringWithCharset(8)
+				log.Debug("Using random test name: ", testName)
 			}
-		}
 
-		if testName == "" {
-			log.Debug("Test Name not provided")
-			testName = StringWithCharset(8)
-			log.Debug("Using random test name: ", testName)
-		}
+			postData := ""
 
-		postData := ""
+			startTime := time.Now()
+			duration, err := time.ParseDuration(testDuration)
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
```

Comment:
Can this be “Invalid test duration. Please use the format xs, xm, xh to specify seconds, minutes, hours”?

Reply:
> Can this be “Invalid test duration. Please use the format xs, xm, xh to specify seconds, minutes, hours”?

Or better can we point to the URL, as we have done in perf -help argument ?

Reply:
Done